software/contrib/harmonic_lfos.py

```diff
@@ -95,26 +160,43 @@ def display_selected_lfo(self):
         
         number = self.divisions[self.selected_lfo]
         x = 2 if number >= 10 else 6
-        oled.text(str(number),x,20,1)
+        oled.text(str(number),x,12,1)
+        
+        lfo_mode = self.modes[self.selected_lfo]
+        if lfo_mode == 0:
+            self.draw_wave('sine')
+        elif lfo_mode == 1:
+            self.draw_wave('saw')
+        elif lfo_mode == 2:
+            self.draw_wave('square')
+        elif lfo_mode == 3:
+            self.draw_wave('off')
+        elif lfo_mode == 4:
+            self.draw_wave('random')
         
-    def calculate_voltage(self, multiplier):
+    def calculate_voltage(self, cv, multiplier):
         """Determine the voltage based on current degree, wave shape, and MAX_VOLTAGE"""
         three_sixty = 360*multiplier
         degree_three_sixty = self.degree % three_sixty
-        if self.mode == 0: #Sin
+        lfo_mode = self.modes[cvs.index(cv)]
+        if lfo_mode == 0: #Sin
             voltage = ((0-(cos(self.rad*(1/multiplier)))+1)) * (MAX_VOLTAGE/2)
-        elif self.mode == 1: #Saw
+        elif lfo_mode == 1: #Saw
             voltage = (degree_three_sixty / (three_sixty)) * MAX_VOLTAGE
-        elif self.mode == 2: #Square
+        elif lfo_mode == 2: #Square
             voltage = MAX_VOLTAGE * (int((degree_three_sixty / (three_sixty)) * MAX_VOLTAGE) < (MAX_VOLTAGE/2))
+        elif lfo_mode == 3: #Off
+            voltage = 0
+        elif lfo_mode == 4: #Random(ish)
+            voltage = ((((0-(cos(self.rad*(1/multiplier)))+1)) * (MAX_VOLTAGE/2)) / 3) + ((((0-(cos(self.rad*(1/(multiplier*2.3))))+1)) * (MAX_VOLTAGE/2)) / 3) + ((((0-(cos(self.rad*(1/(multiplier*5.6))))+1)) * (MAX_VOLTAGE/2)) / 3)
```

Comment:
Optional: This line looks like it has more meaning that just random voltage. Future readers of this code are going to want more explanation of what its doing and why it isn't just using the standard library random. Even just "# Sum of 3 sine waves" will do.